prov/sockets/src/sock_conn.c

```diff
@@ -151,23 +163,24 @@ static struct sock_conn *sock_conn_map_insert(struct sock_ep_attr *ep_attr,
 			return NULL;
 	}
 
-	index = map->used;
+	index = sock_conn_get_next_index(map);
```

Comment:
We can do this check only when the next idx is map->size - 1. That way, we can avoid this O(n) search every time a new connection is inserted.